### PR TITLE
[medPython] Refactoring

### PR DESCRIPTION
**Pull request by fcollot**
This PR makes many changes to medPython. I won't list them all but here are the main ones:

- Removes the explicit linking mechanism (too many problems).

- Adds a Python console accessible with Ctrl+Shift+P or Cmd+Shift+P (I wasn't planning on adding it now but it is very useful for working on the pipeline plugin). This is an experimental feature at the moment.

- Adds two command line options (needs fixing up, doesn't return proper status code):
   **--test-python**
      Runs a series of basic tests on medPython at startup. The results are displayed on the output. An exception does not block execution.
   **--test-python-with-crash**
      Same as above but exceptions are not caught to allow debugging.

- Splits medPython in two sub-projects **medPythonBase** and **medPythonTools** (this was necessary for dependancy reasons).

- Adds the minimum bindings needed to get the console working. I'll add the remaining bindings with the pipeline plugin.

### CODE REVIEW

**Comment by fcollot** [comment]
I forgot to mention this probably doesn't compile in Windows right now.
Aside from that, here is a screenshot of the console (the "help", "license" etc. commands are the default Python ones but we can replace them with ours later):

Screen Shot 2021-09-10 at 15 10 46

**Comment by fcollot** [comment]
(as a proof of concept the console is entirely written in Python using the Qt bindings in medPython : https://github.com/medInria/medInria-public/blob/cac3d5ffe209a5f4bbb317190996ce408308b071/src/layers/medPython/tools/python/console.py)

**Review by juliencastelneau — APPROVED** [review]

**Review by mathildemerle — APPROVED** [review]

File: src/app/medInria/medApplication.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ medApplication::medApplication(int & argc, char**argv) :
     this->setOrganizationDomain("fr");
     this->setWindowIcon(QIcon(":/medInria.ico"));
 
-    medLogger::initialize();
+    //medLogger::initialize();
```
[the remark this answers is not in the document]
Comment (fcollot) [reply src/app/medInria/medApplication.cpp:59]: This must have been a test I did a while back. Didn't notice it. Thanks

[the remark this answers is not in the document]
Comment (fcollot) [reply src/app/medInria/medApplication.cpp]: fixed (plus some spacing on a python file)

**Pull request merged**